.github/workflows/validate.yml

```diff
@@ -0,0 +1,280 @@
+      - name: Install Autotools
+        if: runner.os == 'macOS'
+        run: |
+          brew install automake
+
+      - name: Set validate inputs
+        run: |
+          FLAGS=""
+          if [[ "${{ matrix.cli }}" == "false" ]]; then
+            FLAGS="$FLAGS --lib-only"
+          fi
+          if [[ ${{ matrix.ghc }} == ${{ env.GHC_FOR_SOLVER_BENCHMARKS }} ]]; then
+            FLAGS="$FLAGS --solver-benchmarks"
+          fi
+          if [[ ${{ matrix.ghc }} == ${{ env.GHC_FOR_COMPLETE_HACKAGE_TESTS }} ]]; then
+            FLAGS="$FLAGS --complete-hackage-tests"
+          fi
+          echo "FLAGS=$FLAGS" >> $GITHUB_ENV
+
+      - name: Validate print-config
+        run: sh validate.sh -j 2 -w ghc-${{ matrix.ghc }} -v $FLAGS -s print-config
+
+      - name: Validate print-tool-versions
+        run: sh validate.sh -j 2 -w ghc-${{ matrix.ghc }} -v $FLAGS -s print-tool-versions
+
+      - name: Validate build
+        run: |
+          sh validate.sh -j 2 -w ghc-${{ matrix.ghc }} -v $FLAGS -s build
+
+      - name: Tar cabal head executable
+        if: matrix.cli != 'false' && matrix.ghc == env.GHC_FOR_RELEASE
+        run: |
+          CABAL_EXEC=$(cabal-plan list-bin --builddir=dist-newstyle-validate-ghc-${{ matrix.ghc }} cabal-install:exe:cabal)
+          # We have to tar the executable to preserve executable permissions
+          # see https://github.com/actions/upload-artifact/issues/38
+          if [[ ${{ runner.os }} == 'Windows' ]]; then
+            CABAL_EXEC=$(cygpath $CABAL_EXEC)
+          fi
+          tar -cvf cabal-head.tar -C $(dirname "$CABAL_EXEC") $(basename "$CABAL_EXEC")
+          echo "CABAL_EXEC_TAR=cabal-head.tar" >> $GITHUB_ENV
+
+      # We upload the cabal executable built with the ghc used in the release for:
+      # - Reuse it in the dogfooding job (although we could use the cached build dir)
+      # - Make it available in the workflow to make easier testing it locally
+      - name: Upload cabal-install executable to workflow artifacts
+        if: matrix.cli != 'false' && matrix.ghc == env.GHC_FOR_RELEASE
+        uses: actions/upload-artifact@v2
+        with:
+          name: cabal-${{ runner.os }}-${{ matrix.ghc }}
+          path: ${{ env.CABAL_EXEC_TAR }}
+
+      - name: Validate lib-tests
+        env:
+          # `rawSystemStdInOut reports text decoding errors`
+          # test does not find ghc without the full path in windows
+          GHCPATH: ${{ steps.setup-haskell.outputs.ghc-exe }}
+        run: sh validate.sh -j 2 -w $GHCPATH -v $FLAGS -s lib-tests
+
+      - name: Validate lib-suite
+        run: sh validate.sh -j 2 -w ghc-${{ matrix.ghc }} -v $FLAGS -s lib-suite
+
+      - name: Validate cli-tests
+        if: matrix.cli != 'false'
+        run: sh validate.sh -j 2 -w ghc-${{ matrix.ghc }} -v $FLAGS -s cli-tests
+
+      - name: Validate cli-suite
+        if: matrix.cli != 'false'
+        run: sh validate.sh -j 2 -w ghc -v $FLAGS -s cli-suite
```

Comment:
oops, i've lost the comment about why using raw ghc intead ghc-x.y.z

Reply:
Should we do the same in the other steps too?
Actually, `validate.sh` would use by default `ghc-8.2.2` ([link](https://github.com/haskell/cabal/blob/906a8bcabc307c6c607cf1b9826a6b329e3ceb7f/validate.sh#L7). If we change it to a generic `ghc` we could avoid passing `-w` everywhere.

Reply:
hmm make sense, i wanted to touch validate.sh as little as possible but that default is not sensible anymore

Reply:
done